src/merkle.rs

```diff
@@ -13,7 +13,10 @@ struct MerkleTree {
     leaves: Vec<Hash>,    // This will be uses to add new elements to the tree and rebuild it
 }
 
-type MerkleProof = Vec<Hash>;
+/// Type alias for a Merkle proof
+/// A proof is a list of hashes that can be used to verify the membership of a leaf in the tree
+/// Each hash has associated a boolean value that indicates if the hash is a left or right sibling
+type MerkleProof = Vec<(Hash, bool)>;
```

Comment:
Then it should be an enum `Left`/`Right` I think, which adds more meaning to the value.

Reply:
I second this as well. Enums are our friend! 

Specially Rust enums, super useful feature.

Reply:
Nice, enums will be used for the next PR.